myair_client/rest_client.py

```diff
@@ -0,0 +1,702 @@
+            )
+
+            # The graphql API only works properly if we provide the expected country code
+            # The rest of the parameters are required, but don't seem to be further validated
+            if "myAirCountryId" not in jwt_data:
+                _LOGGER.error(f"myAirCountryId not found in jwt_data")
+                raise ParsingError("myAirCountryId not found in jwt_data")
+            self._country_code = jwt_data["myAirCountryId"]
+            _LOGGER.info(f"Country Code: {self._country_code}")
+        if not self._country_code:
+            _LOGGER.error(
+                "country_code not defined and id_token not present to identify it"
+            )
+            raise ParsingError(
+                "country_code not defined and id_token not present to identify it"
+            )
+        _LOGGER.debug(f"[gql_query] country_code: {self._country_code}")
+
+        graphql_url: str = self._region_config["graphql_url"]
+        headers: dict[str, Any] = {
+            "x-api-key": self._region_config["myair_api_key"],
+            "Authorization": authz_header,
+            # There are a bunch of resmed headers sent to this API that seem to be required
+            # Unsure if this is ever validated/can break things if these values change
+            "rmdhandsetid": "02c1c662-c289-41fd-a9ae-196ff15b5166",
+            "rmdlanguage": "en",
+            "rmdhandsetmodel": "Chrome",
+            "rmdhandsetosversion": "127.0.6533.119",
+            "rmdproduct": self._region_config["product"],
+            "rmdappversion": "1.0.0",
+            "rmdhandsetplatform": "Web",
+            "rmdcountry": self._country_code,
+            "accept-language": "en-US,en;q=0.9",
+        }
+        json_query: dict[str, Any] = {
+            "operationName": operation_name,
+            "variables": {},
+            "query": query,
+        }
+        _LOGGER.debug(f"[gql_query] graphql_url: {graphql_url}")
+        _LOGGER.debug(
+            f"[gql_query] headers: {redact_dict(headers)}"
+        )
+        _LOGGER.debug(
+            f"[gql_query] json_query: {redact_dict(json_query)}"
+        )
+
+        async with self._session.post(
+            graphql_url,
+            headers=headers,
+            json=json_query,
+        ) as records_res:
+            _LOGGER.debug(f"[gql_query] records_res: {records_res}")
+            records_dict: dict[str, Any] = await records_res.json()
+            _LOGGER.debug(
+                f"[gql_query] records_dict: {redact_dict(records_dict)}"
+            )
+            await self._resmed_response_error_check(
+                "gql_query", records_res, records_dict, initial
+            )
+
+        return records_dict
+
+    async def get_sleep_records(self, from_time: datetime, to_time: datetime) -> list[SleepRecord]:
+        end_month = to_time.strftime("%Y-%m-%d")
+        start_month = from_time.strftime("%Y-%m-%d")
+
+        query: str = """query GetPatientSleepRecords {
```

Comment:
Few insignificant remarks:
- f-strings look better then .replace()
- string literals in Python can be wrapped in either double or single quotes.  That's a feature, if one e.g. has double quotes in the string, but does not want to escape them. 

```
query = f'''query ... "{start_month}", endMonth: "end_month", ...'''
'''